### PR TITLE
ref bug #131000 Remaining days off

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Sửa lỗi cập nhật lại số ngày nghỉ thường niên còn lại khi yêu cầu nghỉ phép được accept
Redmine: https://dev.framgia.com/redmine/issues/131000

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        super.onCreate();
+
+        DaggerFireBaseComponent.builder()
+                .appComponent(((MainApplication) getApplication()).getAppComponent())
+                .fireBaseModule(new FireBaseModule(this))
+                .build()
+                .inject(this);
+    }
+
+    @Override
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: WARNING: Method length is 69 lines (max allowed is 60).

File: app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java [hunk 2]
```diff
@@ -169,12 +156,30 @@ public boolean test(OffType offType) throws Exception {
                         return mUserRepository.getListLeaveType();
                     }
                 })
+                .flatMap(
+                        new Function<List<LeaveType>, ObservableSource<BaseResponse<OffTypeDay>>>
+                                () {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java]: ERROR: 'operator new lparen' have incorrect indentation level 32, expected level should be 20, 24.

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java]: ERROR: '(' should be on the previous line.

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                                new Function<List<OffType>,
+                                        ObservableSource<BaseResponse<OffTypeDay>>>() {
+                                    @Override
+                                    public ObservableSource<BaseResponse<OffTypeDay>> apply(
+                                            @NonNull List<OffType> offTypes) throws Exception {
+                                        offTypesCompany = offTypes;
+                                        return mUserRepository.getRemainingDayOff();
+                                    }
+                                })
+                        .observeOn(mBaseSchedulerProvider.ui())
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: ERROR: [PMD-CPD] 14 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+        Uri uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+        NotificationCompat.Builder notificationBuilder =
+                new NotificationCompat.Builder(this).setCategory(NotificationCompat.CATEGORY_PROMO)
+                        .setLargeIcon(BitmapFactory.decodeResource(getResources(),
+                                R.mipmap.ic_launcher_round))
+                        .setAutoCancel(true)
+                        .addAction(R.drawable.ic_view, getString(R.string.view_detail),
+                                pendingIntent)
+                        .setSmallIcon(R.drawable.ic_notification_w)
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: WARNING: [IconColors] Notification icons must be entirely white
 Notification icons and Action Bar icons should only white and shades of gray. See the Android Design Guide for more details. Note that the way Lint decides whether an icon is an action bar icon or a notification icon is based on the filename prefix: `ic_menu_` for action bar icons, `ic_stat_` for notification icons etc. These correspond to the naming conventions documented in http://developer.android.com/guide/practices/ui_guidelines/icon_design.html

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                            new Function<List<OffType>,
+                                    ObservableSource<BaseResponse<OffTypeDay>>>() {
+                                @Override
+                                public ObservableSource<BaseResponse<OffTypeDay>> apply(
+                                        @NonNull List<OffType> offTypes) throws Exception {
+                                    offTypesCompany = offTypes;
+                                    return mUserRepository.getRemainingDayOff();
+                                }
+                            })
+                    .observeOn(mBaseSchedulerProvider.ui())
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: ERROR: [PMD-CPD] 13 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java

File: app/src/main/java/com/framgia/wsm/data/source/UserDataSource.java [hunk 6]
```diff
@@ -45,6 +46,8 @@
 
         Observable<List<OffType>> getListOffType();
 
+        Observable<BaseResponse<OffTypeDay>> getRemainingDayOff();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/source/UserDataSource.java]: pls change ``Observable`` to ``Single``

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            String requestType =
+                    SHOW.equals(notificationType) ? remoteMessage.getData().get(PERMISSION)
+                            : Constant.BLANK;
+            sendNotification(messageResponse, requestType);
+            updateRemainingDayOff(requestType);
+        }
+    }
+
+    private void updateRemainingDayOff(String requestType) {
+        if (REQUEST_OFFS.equals(requestType)) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]:
```java
if (!REQUEST_OFFS.equals(requestType)){
return;
}
....
```

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+                        }
+                    })
+                    .filter(new Predicate<OffType>() {
+                        @Override
+                        public boolean test(@NonNull OffType offType) throws Exception {
+                            return offType.getPayType().equals(COMPANY);
+                        }
+                    })
+                    .toList()
+                    .toObservable()
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: remove 2 line 113 & 114

File: app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+                .toObservable()
+                .flatMap(new Function<List<OffType>, ObservableSource<BaseResponse<OffTypeDay>>>() {
+                    @Override
+                    public ObservableSource<BaseResponse<OffTypeDay>> apply(
+                            @NonNull List<OffType> offTypes) throws Exception {
+                        offTypesCompany = offTypes;
+                        return mUserRepository.getRemainingDayOff();
+                    }
+                })
+                .observeOn(mBaseSchedulerProvider.ui())
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/firebase/FireBaseMessageService.java]: ERROR: [PMD-CPD] 13 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/login/LoginPresenter.java

**Pull request merged**